lib/java/src/org/apache/thrift/transport/TIOStreamTransport.java

```diff
@@ -174,6 +175,8 @@ public int read(byte[] buf, int off, int len) throws TTransportException {
     int bytesRead;
     try {
       bytesRead = inputStream_.read(buf, off, len);
+    } catch (SocketTimeoutException ste) {
+      throw new TTransportException(TTransportException.TIMED_OUT, ste);
```

Comment:
I'm not familiar with the java library code so this might be a stupid question: is it that we already handled `TTransportException` with `TIMED_OUT` in the library code, so translating those `SocketTimeoutException` into `TTransportException` would be enough to resolve the issue reported?

Reply:
Hey @fishy. the code I am proposing does just that.

Translating (wrapping) `SocketTimeoutException` into `TTransportException` :)

Reply:
Without this code change, it gets assigned a value of `UNKNOWN` (see below) and ends up in the logging.

Reply:
you didn't answer the first part of my question: is it that we already handled `TTransportException` with `TIMED_OUT` in the library code? :)

Reply:
https://github.com/apache/thrift/blob/master/lib/java/src/org/apache/thrift/server/TThreadPoolServer.java#L285-L291

This is where the TIMED_OUT is handled differently.